fe/fe-core/src/main/java/org/apache/doris/nereids/rules/expression/rules/FoldConstantRuleOnBE.java

```diff
@@ -264,7 +261,7 @@ private static boolean shouldSkipFold(Expression expr) {
 
         // This kind of function is often used to change the attributes of columns.
         // Folding will make it impossible to construct columns such as nullable(1).
-        if (expr instanceof Nullable || expr instanceof NonNullable) {
+        if (expr.containsType(Nullable.class) || expr.containsType(NonNullable.class)) {
```

Comment:
this will traverse expr tree too many times, could we only traverse once by `expr.anyMatch(e -> shouldSkipFold(e))` ?